### PR TITLE
i18n(ko): Add new `upgrade-astro.mdx` translation

### PR DESCRIPTION
**Pull request by NavyStack**
This PR updates the Korean translation of the `upgrade-astro.mdx` page.

- I have tried to translate as described in RFC2119.
- I have unified the translations for majors, minors, and patches based on the translations in https://semver.org/lang/ko/

@jsparkdev Please check.

- I used the translation of "deprecated" as "사용 중단'
  - Previously, the word was consistently translated as "제거됨".
- I translated "released" as a "릴리즈" for consistency, rather than a "배포".
  - Which is inconsistent with https://semver.org/lang/ko/.

### CODE REVIEW

**Review by dreyfus92 — COMMENTED** [review]
Hey @NavyStack, thanks a lot for your contribution! I'm going to loop in @jsparkdev, who's been working on the Korean translation of `astro-docs`, so he can take a look and share his thoughts.

**Comment by NavyStack** [comment]
Of course @dreyfus92 , I also gave @jsparkdev permission to my fork repository.

File: src/content/docs/ko/upgrade-astro.mdx [hunk 1]
```diff
@@ -0,0 +1,182 @@
+---
+title: Astro 업그레이드하기
```
Comment (jsparkdev) [inline src/content/docs/ko/upgrade-astro.mdx:2]:
```suggestion
title: Astro 업그레이드
```

File: src/i18n/ko/nav.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ export default NavDictionary({
 	'getting-started': '개요',
 	install: '설치',
 	'editor-setup': '편집기 설정',
+	'upgrade-astro': '업그레이드 가이드',
```
Comment (jsparkdev) [inline src/i18n/ko/nav.ts:8]:
```suggestion
	'upgrade-astro': 'Astro 업그레이드',
```

File: src/content/docs/ko/upgrade-astro.mdx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+이 가이드에서는 Astro 버전 및 관련 종속성을 업데이트하는 방법, 한 버전에서 다음 버전으로 변경된 내용을 알아보는 방법, Astro의 버전 관리 시스템 및 해당 문서 업데이트를 이해하는 방법에 대해 설명합니다.
+
+## 달라진 점은 무엇인가요?
+현재 Astro의 최신 릴리스는 <Version pkgName="astro" /> 입니다.
+
+모든 변경사항의 전체 목록은 [Astro의 변경 로그](https://github.com/withastro/astro/blob/main/packages/astro/CHANGELOG.md)에서, 새로운 [주버전 버전](#주버전-변경)으로 업그레이드하는 데 필요한 중요한 지침은 [업그레이드 가이드](#업그레이드-가이드)에서 확인할 수 있습니다.
+
+
+## 최신 버전으로 업그레이드하기
```
Comment (jsparkdev) [inline src/content/docs/ko/upgrade-astro.mdx:18]:
```suggestion
## 최신 버전으로 업그레이드
```

File: src/content/docs/ko/upgrade-astro.mdx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+  <Fragment slot="yarn">
+  ```shell
+  # Astro와 공식 통합을 함께 업그레이드하기
+  yarn dlx @astrojs/upgrade
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+
+### 수동으로 업그레이드하기
```
Comment (jsparkdev) [inline src/content/docs/ko/upgrade-astro.mdx:44]:
```suggestion
### 수동으로 업그레이드
```

File: src/content/docs/ko/upgrade-astro.mdx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  # 예시: React 및 Tailwind 통합을 포함해서 Astro를 업그레이드 합니다
+  yarn add astro@latest @astrojs/react@latest @astrojs/tailwind@latest
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+### 특정 버전을 설치하기
```
Comment (jsparkdev) [inline src/content/docs/ko/upgrade-astro.mdx:69]:
```suggestion
### 특정 버전 설치
```

File: src/content/docs/ko/upgrade-astro.mdx [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+Astro의 기본 문서 페이지는 항상 **최신 출시된 버전의 Astro에 대해 정확합니다**. 이전 버전의 작동 방식을 설명하거나 비교하지 않으며, 업데이트되거나 변경된 동작을 강조하지도 않습니다.
+
+아래 업그레이드 가이드에서 새 버전과 이전 버전을 비교하여 변경 사항에 대한 설명을 확인하세요. 업그레이드 가이드에는 변경 사항, 사용 중단, 기능 제거 및 대체, 업데이트된 사용 안내 등 코드를 변경해야 할 수 있는 모든 내용이 포함되어 있습니다. Astro의 각 변경 사항에는 프로젝트 코드를 성공적으로 업데이트하는 데 도움이 되는 "어떻게 해야 하나요?" 섹션이 포함되어 있습니다.
+
+- [v4로 업그레이드](/ko/guides/upgrade-to/v4/)
+- [v3로 업그레이드](/ko/guides/upgrade-to/v3/)
+- [v2로 업그레이드](/ko/guides/upgrade-to/v2/)
+- [v1로 업그레이드](/ko/guides/upgrade-to/v1/)
+
+### 이전 문서(유지 관리되지 않는 문서)
```
Comment (jsparkdev) [inline src/content/docs/ko/upgrade-astro.mdx:113]:
```suggestion
### 이전 문서 (관리되지 않음)
```

File: src/content/docs/ko/upgrade-astro.mdx [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+- 향후 주버전 릴리스에서 제거될 것이라는 경고와 함께 기존 기능/옵션의 **사용 중단**이 표시됩니다.
+- 새로운 기능 도입.
+- 통합 후크에 새로운 옵션 도입.
+- 새로운 어댑터를 만드는 데 주로 사용되는 `astro/app`에 새로운 기능 도입.
+
+부버전 릴리스에는 소규모 수버전 변경 사항도 동시에 포함될 수 있습니다.
+
+### 주버전 변경
+
+주버전 릴리스에는 최소한 일부 기존 코드에 대한 획기적인 변경 사항이 포함됩니다. 이러한 중요한 변경 사항은 항상 Astro의 ["vX로 업그레이드" 가이드](#upgrade-guides)에 문서화되어 있습니다.
```
Comment (jsparkdev) [inline src/content/docs/ko/upgrade-astro.mdx:161]:
```suggestion
주버전 릴리스에는 최소한 일부 기존 코드에 대한 획기적인 변경 사항이 포함됩니다. 이러한 중요한 변경 사항은 항상 Astro의 ["vX로 업그레이드" 가이드](#업그레이드-가이드)에 문서화되어 있습니다.
```

File: src/content/docs/ko/upgrade-astro.mdx [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+
+## 문서 업데이트
+
+이 문서는 [부버전 릴리스](#부버전-변경) 및 [주버전 버전 릴리스](#주버전-변경)가 있을 때마다 업데이트됩니다. 새로운 기능이 추가되거나, 기존 사용법이 변경되면, 문서가 업데이트되어 Astro의 **현재 동작**을 반영합니다. 프로젝트가 업데이트되지 않은 경우, 일부 동작이 최신 문서와 일치하지 않을 수 있습니다.
+
+새로운 기능은 추가된 특정 버전 번호와 함께 문서에 추가됩니다. 즉, 최신 버전의 Astro로 업데이트하지 않은 경우 일부 문서화된 기능을 사용하지 못할 수 있습니다. 새로운 기능을 사용하기 전에 항상 `추가됨:` 버전 번호를 확인하고 프로젝트가 업데이트되었는지 확인하세요!
```
Comment (jsparkdev) [inline src/content/docs/ko/upgrade-astro.mdx:96]:
```suggestion
새로운 기능은 추가된 특정 버전 번호와 함께 문서에 추가됩니다. 즉, 최신 버전의 Astro로 업데이트하지 않은 경우 일부 문서화된 기능을 사용하지 못할 수 있습니다. 새로운 기능을 사용하기 전에 항상 `Added in:` 버전 번호를 확인하고 프로젝트가 업데이트되었는지 확인하세요!
```

Comment (jsparkdev) [reply src/content/docs/ko/upgrade-astro.mdx:96]: 버전 번호를 나타내기 위해 사용되는 `<Since />` 컴포넌트는 번역되지 않은 문자열인 `Added in:` 을 렌더링합니다.

**Review by jsparkdev — APPROVED** [review]

**Comment by jsparkdev** [comment]
LGTM

**Comment by NavyStack** [comment]
@jsparkdev 고생 많으셨습니다 :)

**Review by yanthomasdev — APPROVED** [review]
LGTM! Good job 🙌

**Pull request merged**